lib/galaxy/webapps/galaxy/api/library_contents.py

```diff
@@ -209,9 +212,16 @@ def create( self, trans, library_id, payload, **kwd ):
 
         # are we copying an HDA to the library folder?
         #   we'll need the id and any message to attach, then branch to that private function
-        from_hda_id, ldda_message = ( payload.pop( 'from_hda_id', None ), payload.pop( 'ldda_message', '' ) )
-        if create_type == 'file' and from_hda_id:
-            return self._copy_hda_to_library_folder( trans, from_hda_id, library_id, real_folder_id, ldda_message )
+        from_hda_id, from_hdca_id, ldda_message = ( payload.pop( 'from_hda_id', None ), payload.pop( 'from_hdca_id', None ), payload.pop( 'ldda_message', '' ) )
+        log.debug(payload)
+        if create_type == 'file':
+            rval = []
+            if from_hda_id:
+                rval.append(self._copy_hda_to_library_folder( trans, self.hda_manager, self.decode_id(from_hda_id), real_folder_id, ldda_message ))
```

Comment:
I was thinking it would be better to consistently return a list, but I can change this to unpack the list if the length of the list is 1. I don't have a preference either way, just let me know!